### DISCUSSION TITLE
File types available for references (can't reference to yaml file)

### DISCUSSION THREAD

**Discussion by villanella** [strictdoc-project/strictdoc#1323] [Q&A] [answered]
Hello! How can I reference a requirement to yaml source code (for example, Ansible roles etc)?
It seems that yaml extension is excluded, because I get the error "requirement references a file that does not exist" when I reference to yaml file. Where is the setting of excluded extensions?

I've added the source dir to strictdoc.toml, but it doesn't help:
```
include_source_paths = [                                                                                                                          
   "tests/**",
   "ansible/**"
]
```
If I change file extension from '.yaml' to '.py', the reference works. But it's not a workaround.

**Comment by villanella** [comment]
I've added YamlLexer by analogy with others, and it works!:

added to core/finders/source_files_finder.py:
```
class SourceFileType(Enum):
...
    YAML = [".yaml", ".yml"]
...
    def is_yaml_file(self):
        return self.file_type == SourceFileType.YAML
```

and to export/html/generators/source_file_view_generator.py:
```
from pygments.lexers.data import YamlLexer
...

    def get_pygmented_source_lines(
...
        elif source_file.is_yaml_file():
            lexer = YamlLexer()
...

```

**Reply by stanislaw** [reply]
That's pretty much what I was going to say, and you did it right!

I would gladly accept a PR with this change.

**Reply by stanislaw** [reply, marked as answer]
Closing by #1324.

**Comment by villanella** [comment]
@stanislaw please, see my PR https://github.com/strictdoc-project/strictdoc/pull/1324
Should I create an issue?

**Reply by stanislaw** [reply]
No, the issue is not needed. I will check the PR during the day.
